pulp-glue/pulp_glue/common/openapi.py

```diff
@@ -504,20 +504,19 @@ def render_request_body(
                 elif content_type.startswith("application/x-www-form-urlencoded"):
                     data = body
                 elif content_type.startswith("multipart/form-data"):
-                    uploads: Dict[str, UploadType] = {}
+                    uploads: Dict[str, Tuple[str, UploadType, str]] = {}
                     data = {}
                     # Extract and prepare the files to upload
                     if body:
                         for key, value in body.items():
-                            if isinstance(value, (bytes, BufferedReader)):
-                                uploads[key] = value
+                            if isinstance(value, bytes):
+                                uploads[key] = (key, value, "application/octet-stream")
+                            elif isinstance(value, BufferedReader):
+                                uploads[key] = (value.name, value, "application/octet-stream")
```

Comment:
I don't fine the name attribute in the docs for BufferedReader (i'm also not sure this is the best way to determine we are transferring a file.) How about we do
```suggestion
                                uploads[key] = (getattr(value, "name", key), value, "application/octet-stream")
```
instead of the new if and then look if we can use the openapi-spec to see whether a file is supposed to be transferred?

Reply:
What do you mean by check the api-spec to see if a file is supposed to be transferred?